src/include/optimizer/abstract_optimizer.h

```diff
@@ -45,8 +47,14 @@ class AbstractOptimizer {
 
   virtual std::shared_ptr<planner::AbstractPlan> BuildPelotonPlanTree(
       const std::unique_ptr<parser::SQLStatementList> &parse_tree, concurrency::Transaction *txn) = 0;
+  
+  inline void SetDefaultDatabaseName(std::string default_database_name) {
+      this->default_database_name_ = default_database_name;
+  }
 
   virtual void Reset(){};
+
+  std::string default_database_name_ = DEFAULT_DB_NAME;
```

Comment:
I don't think we should do this. This is just going to lead to everything landing in DEFAULT_DB_NAME. Let's remove this here and have `default_database_name` be passed in to `AbstractOptimizer`'s constructor.

@hzxa21 Thoughts?

Reply:
The optimizer doesn't need `default_database_name` previously. I think the reason why Tianyi needs to add this into the optimizer is that he changes the `GetDatbaseName()`  function signature in all AST node to be `GetDatbaseName(string default_db_name)` and the optimizer needs to call these functions. However, I think it is not necessary to do that because in the binder, the default db name should already be filled in if the db filed is empty. Correct me if I miss something @ChTimTsubasa.

Reply:
I agree with @hzxa21. It sounds like we do not need to expose this aspect of the API to the optimizer. The optimizer does not need a default DB name.

Reply:
@hzxa21 Where should I retrieve the bound database name after binding?

Reply:
Shouldn't it be in the AST node?

Reply:
@apavlo I talked with @hzxa21 about this problem. The solution would be to try set the table_info of AST node by passing in the default database name. Previously I just return the database name back without modifying the AST node.